backend/src/case_file/case_file.service.ts

```diff
@@ -555,6 +575,149 @@ export class CaseFileService {
+  {
+    try
+    {
+      let caseFileId: string;
+      await this.prisma.$transaction(async (db) => {
+        //create case
+        const caseFile = await db.case_file.create({
+          data: {
+            agency_code: {
+              connect: {
+                agency_code: reviewInput.agencyCode
+              }
+            },
+            create_user_id: reviewInput.userId,
+            create_utc_timestamp: new Date(),
+            review_required_ind: true,
+            case_code_case_file_case_codeTocase_code: {
+              connect: {
+                case_code: reviewInput.caseCode
+              }
+            },
+          }
+        });
+        caseFileId = caseFile.case_file_guid
+        //create lead
+        await db.lead.create({
+          data: {
+            lead_identifier: reviewInput.leadIdentifier,
+            case_identifier: caseFile.case_file_guid,
+            create_user_id: reviewInput.userId,
+            create_utc_timestamp: new Date()
+          }
+        });
+      });
+      return caseFileId;
+    }
+    catch (err) {
+      console.error(err);
+      throw new GraphQLError('Error in createReviewCase', {});
+    }
+  }
+
+  //Create review complete action in table action
+  async createReviewComplete(reviewInput: ReviewInput): Promise<string>
+  {
+    try
+    {
+      let actionId: string;
+      await this.prisma.$transaction(async (db) => {
+        let actionTypeActionXref = await db.action_type_action_xref.findFirstOrThrow({
+          where: {
+            action_type_code: "CASEACTION",
+            action_code: "COMPLTREVW"
+          },
+          select: {
+            action_type_action_xref_guid: true
+          }
+        });
+        const reviewAction = await db.action.create({
+          data: {
+            case_guid: reviewInput.caseIdentifier,
+            action_type_action_xref_guid: actionTypeActionXref.action_type_action_xref_guid,
+            actor_guid: reviewInput.reviewComplete.actor,
+            action_date: reviewInput.reviewComplete.date,
+            active_ind: true, //True: review complete, false: review not complete
+            create_user_id: reviewInput.userId,
+            create_utc_timestamp: new Date
+          }
+        });
+        actionId = reviewAction.action_guid
+      });
+      return actionId;
+    }
+    catch (err) {
+      console.error(err);
+      throw new GraphQLError('Error in createReviewComplete', {});
+    }
+  }
+
+  //Mutation createReview
+  async createReview(reviewInput: ReviewInput) {
+    try {
+      let result = {
+        ...reviewInput
+      }
+      //If case is not exists -> create case
+      if(!reviewInput.caseIdentifier) {
+        const caseFileId = await this.createReviewCase(reviewInput);
+        result.caseIdentifier = caseFileId
+        result.isReviewRequired = true
+      }
+      //Else update review_required_ind
+      else {
+        await this.prisma.$transaction(async (db) => {
```

Comment:
you should remove this transaction boundary here as you're not actually making any changes in this function. To select the caseFile you're better off using ```this.prisma.case_file``` instead